docs/deployment/microbit-deployment-draft.md

```diff
@@ -0,0 +1,166 @@
+
+{<< what other things should constitute a breaking change, and should we allow certain kinds of breaking change inside a version of the editor? IE can we support more than one MicroPython version for a particular deployed version of the front-end?>>}}
+
+It may be desirable to agree a high water mark in memory and growing beyond this is considered a breaking change. It would then be necessary to police memory growth but people do already do this I think. This would allow small increases in memory usage to be possible without a breaking change.
+
+### URL versioning scheme
+
+For each published major version, the deployed editor will live at
+
+`python.microbit.org/v<major version>/editor.html`
+
+Because the use of semantic versioning ensures that things will not break for the user when upgrading minor versions, there will not be fine-grained control over which minor version a user gets. 
+
+For example visiting `python.microbit.org/v2/` will serve version `2.<latest minor>`
+
+### Versions in URLs
+
+When a script is published or shared, it must be published including the major version of the editor (and therefore the MicroPython stack) that it was created with so that anyone following the link will be taken back to the version the script was created with.
+
+If a user visits python.microbit.org/latest/ and is working on the editor there, the URL published should still be versioned, such as python.microbit.org/v1/. 
+
+See “workflows” for discussion of how the user will be notified about newer versions
+
+
+### Versioning in hex files
+
+When the user is provided a hex file, the full semantic editor version must also be embedded into the hex file at a known location so that when unpacking the hex, any version of the editor can offer to load the version that the script was created in.
+
+The hex file must also contain some editor unique identity, so that any hex can be related to the appropriate editor, perhaps automatically (e.g. MP signature at known location).This could be part of the version number?
+
+See “workflows” for discussion of how the user will be notified about the fact they are heading to an older version.
+
+(This kind of thing can enable the ability to show a user the exact version of the source code that was used to generate their hex file, including MicroPython code)
+
+## Release cadence
+
+Because we must commit to maintaining and fixing versions that are deployed on python.microbit.org it is desirable to make a small number of major release. 
+
+We propose an annual cadence for major releases, with an RC released early in the year so that a stable release can be made in July after the school year ends. This gives teachers time to become familiar with it before the school year start. The previous year’s version will remain available after this but will no longer be redirected to via python.microbit.org/latest or python.microbit.org/
```

Comment:
I don't like this at all. 

While I understand the need for stability, it means development is constrained to a glacial pace of release. Furthermore, it'll encourage others to just update to the latest version and host it themselves.

Why not make the cadence three times a year to reflect the cadence of the school year: Christmas holidays, Easter and Summer holidays.

I want to stress that the wider Python community won't follow these cadences (herding cats etc...). This will be a major source of pain for *me* personally as people tap my shoulder asking where / when their contribution / fix will land. 

Once a year also means all the effort and rush for a stable release happens over the summer when people want to go on holiday.

My biggest concern is the King Canute argument - you may say you're going to do X, Y and Z (and *I* may agree with you on such points) but if the wider community doesn't agree they'll just find ways around them and/or fork, thus making it hard to coordinate efforts. I.e. the tide will come in no matter what you say. Welcome to my world!